l1-contracts/test/fees/MinimalFeeModel.sol

```diff
@@ -3,21 +3,52 @@
 pragma solidity >=0.8.27;
 
 import {FeeMath, OracleInput} from "@aztec/core/libraries/FeeMath.sol";
+import {Timestamp, TimeFns, Slot} from "@aztec/core/libraries/TimeMath.sol";
+import {Vm} from "forge-std/Vm.sol";
 
-contract MinimalFeeModel {
+struct BaseFees {
+  uint256 baseFee;
+  uint256 blobFee;
+}
+
+// This actually behaves pretty close to the slow updates.
+struct L1BaseFees {
+  BaseFees pre;
+  BaseFees post;
+  Slot slotOfChange;
+}
+
+struct DataPoint {
+  uint256 provingCostNumerator;
+  uint256 feeAssetPriceNumerator;
+}
+
+contract MinimalFeeModel is TimeFns {
   using FeeMath for OracleInput;
   using FeeMath for uint256;
 
-  struct DataPoint {
-    uint256 provingCostNumerator;
-    uint256 feeAssetPriceNumerator;
-  }
+  // This is to allow us to use the cheatcodes for blobbasefee as foundry does not play nice
+  // with the block.blobbasefee value if using cheatcodes to alter it.
+  Vm internal constant VM = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+  // The L1 base fees are "fixed" for a period of 5 slots.
+  // and is lagging 5 slots behind the current slot.
```

Comment:
lagging "2" slots behing?

Reply:
Never adding numbers in comments again 💀